lib/declarations.d.ts

```diff
@@ -0,0 +1,13 @@
+declare module "resin-discoverable-services" {
+    export interface publishOptions {
+        identifier: string;
+        name: string;
+        host?: string;
+        port: number;
+    }
+
+    export function setRegistryPath(path: string): void;
+    export function enumerateServices(callback?: (error: Error, results: [string]) => void): Promise<[string]>;
+    export function findServices(services: [string], timeout?: number, callback?: (error: Error, results: [string]) => void): Promise<[string]>;
+    export function publishServices(services: [publishOptions], callback?: (error: Error) => void): Promise<void>;
```

Comment:
You want`string[]`, not `[string]`.

`string[]` literally says "this is an array of strings".

`[string]` says "this is a tuple containing one string element". [Tuple syntax](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple) lets you write types like `[number, string, string]`. Here you've used just one element, and TypeScript then lets you go outside the tuple indexes later and uses the union of the tuple types (which here is just `string`). That all means that, yes, this does technically act the same as an array of strings, but it's a pretty weird way of describing that.